operator/internal/manifests/storage/configure.go

```diff
@@ -186,29 +186,58 @@ func ensureObjectStoreCredentials(p *corev1.PodSpec, opts Options) corev1.PodSpe
 			},
 		}
 	case lokiv1.ObjectStorageSecretS3:
-		storeEnvVars = []corev1.EnvVar{
-			{
-				Name: EnvAWSAccessKeyID,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
+		sts := opts.S3 != nil && len(opts.S3.RoleArn) != 0
+		if !sts {
+			storeEnvVars = []corev1.EnvVar{
+				{
+					Name: EnvAWSAccessKeyID,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: KeyAWSAccessKeyID,
 						},
-						Key: KeyAWSAccessKeyID,
 					},
 				},
-			},
-			{
-				Name: EnvAWSAccessKeySecret,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
+				{
+					Name: EnvAWSAccessKeySecret,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: KeyAWSAccessKeySecret,
 						},
-						Key: KeyAWSAccessKeySecret,
 					},
 				},
-			},
+			}
+		} else {
+			storeEnvVars = []corev1.EnvVar{
+				{
+					Name: EnvAWSRoleArn,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: KeyAWSRoleArn,
+						},
+					},
+				},
+				{
+					// TODO (JoaoBraveCoding) fix
+					Name: EnvAWSWebIdentityTokenFile,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: KeyAWSAccessKeySecret,
+						},
+					},
```

Comment:
Since we will never expose setting the value of this to the user, we should consider setting up the value with a helper function when mounting the coresponding serviceaccount token projection, i.e.

1. On plain Kubernetes the token mount looks like this:
```yaml
---
serviceAccountName: lokistack-dev
---
volumeMounts:
# Watch the path is is over the openshift subdir
- mountPath: /var/run/secrets/kubernetes/serviceaccount 
  name: bound-sa-token
---
volumes:
- name: bound-sa-token
  projected:
    sources:
      - serviceAccountToken:
          # Watch out the audience is set to api here
          audience: api        
          expirationSeconds: 3600
          path: token
```
2. On OpenShift  the token mount looks like this:
```yaml
---
serviceAccountName: lokistack-dev
---
volumeMounts:
# Watch the path is is over the openshift subdir
- mountPath: /var/run/secrets/openshift/serviceaccount 
  name: bound-sa-token
---
volumes:
- name: bound-sa-token
  projected:
    sources:
      - serviceAccountToken:
          # Watch out the audient is openshift here *NOT* api
          audience: openshift         
          expirationSeconds: 3600
          path: token
```

Reply:
Added audience to the configuration as an optional parameter in the configMap and I'm thinking of passing down to `ExtractSecret`  the concept of an openshift or k8s cluster so we can correctly set the mountPath.

Thinking about it we can also make the path agnostic no? `/var/run/secrets/wif/serviceaccount` I'm picking WIF as the common denomination for this feature but we can pick something else